src/web/client/services/NPSService.ts

```diff
@@ -19,21 +21,29 @@ export class NPSService{
     }
 
     public static async getEligibility()  {
-        const baseApiUrl = "https://ces-int.microsoftcloud.com/api/v1"; // TODO: change int to prod based on the query params from Studio
-        const accessToken: string = await npsAuthentication(CESSurvey.AUTHORIZATION_ENDPOINT);
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-        const parsedToken = jwt_decode(accessToken) as any;
-        const apiEndpoint = `${baseApiUrl}/${CESSurvey.TEAM_NAME}/Eligibilities/${CESSurvey.SURVEY_NAME}?userId=${parsedToken?.oid}&eventName=${CESSurvey.EVENT_NAME}&tenantId=${parsedToken.tid}`;
-        const requestInitPost: RequestInit = {
-            method: 'POST',
-            body:'{}',
-            headers:NPSService.getCesHeader(accessToken)
-        };
-        const response = await fetch(apiEndpoint, requestInitPost);
-        const result = await response?.json();
-        // TODO: telemetry
-        if( result?.eligibility){
-          WebExtensionContext.setNPSEligibility(true);
+        const region = WebExtensionContext.urlParametersMap?.get(queryParameters.REGION)
+        const baseApiUrl = region === 'test' ? "https://ces-int.microsoftcloud.com/api/v1": "https://ces.microsoftcloud.com/api/v1";
+        
+        try{
+                const accessToken: string = await npsAuthentication(SurveyConstants.AUTHORIZATION_ENDPOINT);
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                const parsedToken = jwt_decode(accessToken) as any;
+                WebExtensionContext.setUserId(parsedToken?.oid)
+                const apiEndpoint = `${baseApiUrl}/${SurveyConstants.TEAM_NAME}/Eligibilities/${SurveyConstants.SURVEY_NAME}?userId=${parsedToken?.oid}&eventName=${SurveyConstants.EVENT_NAME}&tenantId=${parsedToken.tid}`;
+                const requestInitPost: RequestInit = {
+                    method: httpMethod.POST,
+                    body:'{}',
+                    headers:NPSService.getCesHeader(accessToken)
+                };
+                const requestSentAtTime = new Date().getTime();
+                const response = await fetch(apiEndpoint, requestInitPost);
+                const result = await response?.json();
+                if( result?.eligibility){
+                    powerPlatformExtensionContext.telemetry.sendAPISuccessTelemetry(telemetryEventNames.NPS_USER_ELIGIBLE, "NPS Api",httpMethod.POST,new Date().getTime() - requestSentAtTime);
```

Comment:
This is wrong. All you need to do it  WebExtensionContext.telemetry.sendAPISuccessTelemetry

Reply:
fixed